### DISCUSSION TITLE
xdrip+ watchfaces arre not working on GW4 wearos4

### DISCUSSION THREAD

**Discussion by Npavkov** [NightscoutFoundation/xDrip#3126] [Bug reports]
I am using the newest nightly version of xdrip+  and I had to reinstall it on my phone. After the reinstall, I can not get the watch faces on my watch to communicate with my phone version of xdrip+....no data gets sent (although the status of queues on phone app show the bg is being sent to smatwatch aps)......is this a bug?

**Comment by Navid200** [comment]
@psonnera

**Comment by psonnera** [comment]
Did you also uninstall and reinstall xDrip+ on the wztch?

**Comment by Npavkov** [comment]
Yes, I did AFTER I reset the Watch. The watchface seemed to work before resetting and reinstalling the watch app, and reinstalling the phone app, but not after!

**Comment by Npavkov** [comment]
Patrick, are you able to help me figure out why the watchfaces are not working? I would appreciate any help you can give me.

**Reply by psonnera** [reply]
Allright. So, you confirm that you have the xDrip+ wear extension installed on the smartwatch (you can select the watchface) using a sideloading app like Wear installer 2. The watchface opens and you don't see data. You enabled wear in xDrip+ but not the standalone collector mode. Right?

**Comment by Npavkov** [comment]
all correct!!

**Comment by Npavkov** [comment]
on the system status/cloud upload screen I see the BG record being sent to API wear component. It is like the the Phone app and the watch app just cant talk. I also run juggluco which does successfully transfer data from watch to phone.....

**Comment by Npavkov** [comment]
I have a suspicion that the signatures do not match between the watch and the phone apps for xdrip. Is there a way to check the signatures match?

**Reply by psonnera** [reply]
They should as long as you used the same version. When you update the phone xDrip+ it might break the link. https://github.com/NightscoutFoundation/xDrip/issues/1648#issuecomment-818600901

**Comment by Npavkov** [comment]
I have confirmed they are the same version. However I am unable to confirm that the watch record is even being sent from phone xdrip app to watch. I only see nightscout record in packet captures and in the cloud upload queues being sent so it does not look like phone app is even sending the watch api record at all. I see nothing in the upload queues for watch and I also see nothing in log indicating the wearos record is even being sent.  any ideas you or anyone else could offer?  Are there other GW4 users running wearos4 that is having the same issues?

**Comment by Npavkov** [comment]
is there a specific socket (port) that xdrip+ uses to send the data to the watch face? is there another log available  in xdrip to determine if the data is being sent to smart watch interface?

**Reply by psonnera** [reply]
You're right. I don't see any event in the log related to Wear. Maybe they can be added manually but I don't know the keys related to this... Also, did you try to update the watch database (cloud icon) 
![image](https://github.com/NightscoutFoundation/xDrip/assets/33809361/856f274e-f09a-47fc-9903-363a4bd99ca8)
Also, from the phone app, three dots menu, reset wear database?

**Comment by Npavkov** [comment]
ok, according to this logcat, it looks like the watch wear bg reading is attempted by the phone app, but not making it to watch face.????? is there something else I am missing?
[phonelog.txt](https://github.com/NightscoutFoundation/xDrip/files/12886582/phonelog.txt)

**Reply by psonnera** [reply]
I see xDrip+ happily sending data. Next step might be logcat from watch? Also, are why you trying to upload to Dexcom share? (unrelated)

**Comment by Npavkov** [comment]
By the way, I am not able to get the 3 dots menu to come up on the watch face at all/?

**Comment by Npavkov** [comment]
here you go.....
[watchlog.txt](https://github.com/NightscoutFoundation/xDrip/files/12887246/watchlog.txt)

**Reply by psonnera** [reply]
Well, looks like G-Watch is interfering. Use one or the other...

**Comment by Npavkov** [comment]
I have removed gwatch and all other watchface software from phone and watch> rebooted both watch and phone> >>>>Still no data on xdrip watchface????

**Reply by psonnera** [reply]
Try to uninstall and reinstall the xDrip+ watchface

**Comment by Npavkov** [comment]
Already did that multiple times

**Comment by Npavkov** [comment]
new log
[watchlog3.txt](https://github.com/NightscoutFoundation/xDrip/files/12907084/watchlog3.txt)

**Reply by psonnera** [reply]
There still are pieces of the previous installs.
`MARsmini_FreecessController: freezePackage tk.glucodata uid:10168 LEV`
I'd reset the watch and restart from scratch.
I don't understand why this
`WindowManager: Setting visibility of Window{3b234c1 u0 com.eveningoutpost.dexdrip.Home}: false`
and this
`InputDispatcher: Channel [Gesture Monitor] quickpanel (server) is stealing touch from [8bb8105 com.samsung.android.wearable.sysui/com.google.android.clockwork.sysui.mainui.activity.SysUiActivity (server), 3b234c1 com.eveningoutpost.dexdrip.Home (server)]`
Are you using an xDrip+ watchface or a complication?
I would also make sure xDrip+ has rights to be active in background.
`ActivityManager: Stopping service due to app idle: u0a166 -48m3s144ms com.dexdrip.stephenblack.nightwatch/.DataCollectionService`

**Comment by Npavkov** [comment]
Patrick, The way I have my system setup follows:

 

I run Juggluco (tk.glucodata) on phone and watch . The watch is the collector of the bg Bluetooth data from the Libre2 sensor. Juggluco sends the bg reading from the watch to the phone Juggluco app by means of a MIRROR.

The Juggluco on phone receives the reading from the MIRROR and sends bg reading to xdrip via libre Patched app. Then xdrip then calibrates the reading (because Libre2 RAW reading is ALWAYS significantly Lower reading than ACTUAL) and sends reading to Nightscout (for reporting), and the xdrip watchface for a more accurate bg reading on the watch display.

 

The com.eveningoutpost.dexdrip.Home reference is generated from juggluco. So I would expect that in the log.

 

This setup worked consistently  on Wearos 3 (before the watch update to wear 4 came down.). sometime after the update I started having issues with the xdrip watch face not updating. I currently have loaded the Nightscout watchface to the watch in order to get the calibrated bg reading to display on the watch however that updates the watchface ONLY every 7 – 10 mins. Xdrip watchface has always updated every minute when the sensor readings come in.

 

I know you are going to say I should use either Juggcluco or xdrip and not both. However I use Juggluco so my raw readings are sent to LIBREVIEW for my doctors, and I use xdrip because of having to do some kind of calibration on the raw readings to make them more accurate and I use Nightscout for reporting and history of more accurate readings.

 

I really would like to get the xdrip watchface functioning again to display the more accurate readings on my watch. Is there a setting or something on xdrip to identify why the data is not being sent to xdrip watchface? Using a packet analyser on the phone I was NOT able to see the wear api record being sent even though I WAS ABLE to see the Nightscout API record being sent over the network, which almost indicates the issue is with the data being sent out from the xdrip phone app for weor API!! Is there anyone else experiencing this issue with the xdrip watchface on a galaxy watch 4 running wearos 4 that you know of??

**Reply by psonnera** [reply]
Thanks, now it's clearer. As far as I know xDrip+ is active on the watch only when the watchface is selected. Juggluco on the other hand might be able to run in background, but I'm not sure. I suspect Wear OS 4 has changed that somehow. I haven't heard of anybody else doing that. I believe you should stick to Juggluco watchface and have xDrip+ send calibrated data to Nighscout. A sensor tht doesn't read correctly should be replaced. We cruelly lack xDrip+ developers on Wear OS, so I'm not sure anybody can help...

**Comment by Npavkov** [comment]
Thanks for all your help so far. Is there any documentation on the data intents that xdrip+ sends to the watchface??
